### DISCUSSION TITLE
Base32 encoded token secret support

### DISCUSSION THREAD

**Discussion by ilyabreev** [akopachov/flipper-zero_authenticator#43] [Q&A] [answered]
Hi there!

Is there any way to use base32 encoded secret got from `otpauth://totp/smth?secret=xxxxxxx&issuer=zzz` URL
Tried to paste it in config. App runs smoothly but generated codes are wrong.

**Comment by akopachov** [comment]
Hey, 
what you need to take out of such a URIs is whatever is going after `secret=` and till `&` or end of string.

So in your example `otpauth://totp/smth?secret=xxxxxxx&issuer=zzz` what you need to provide to Flipper Authenticator is `xxxxxxx` only.

**Comment by ilyabreev** [comment]
Thanks for your reply.
Yeah that's exactly what i've done. Also I checked that flipper clock is synced through BT with flipper app.
Trying to login into my Github profile for example. I'm using SHA-1.
Still doesn't work.

I wonder if codes generated with flipper must exactly match with Google Authenticator codes?

**Comment by akopachov** [comment]
Have you set correct timezone offset in Flipper Authenticator?

**Comment by ilyabreev** [comment]
Yes. It's MSK for my case (+3h).

**Comment by ilyabreev** [comment]
Forgot to mention that flipper-zero authenticator refreshes codes slightly later than Google Authenticator app (with 0.5-1.5 seconds delay). May be this an issue?

**Reply by akopachov** [reply]
This is due to known problem of clock drift in Flipper. And that's why it is necessary to sync its clock from time to time. 0.5-1.5 sec drift is fine

**Comment by akopachov** [comment, marked as answer]
Just tried to add secret from github myself using CLI and it seems to be working just fine. Can you try it via CLI?

**Reply by akopachov** [reply]
Also tried to manually copy&paste secret straight to a config file and it works just fine as well

**Reply by ilyabreev** [reply]
It worked with CLI! Thank you very much! It's much more convenient than manually adding a secret with flipper keyboard.

**Reply by akopachov** [reply]
Glad it works for you.
That flipper keyboard is kinda obsolete. Originally it was the first input method I added, but then when all other methods were added I realized that highly-unlikely somebody will be ever using it. So I think I'll just retire it one day and completely drop that "add new token" ui.

**Reply by akopachov** [reply]
Tried to enter secret via "add new token" ui and it worked for me too. But I highly recommend to use other methods as entering token via app ui is super-boring and chance of making mistake is very high.

**Comment by pankiller87** [comment]
Is there any possibility to reset authenticator? Probably i made some password and i cannot log in

**Reply by akopachov** [reply]
Do you mean resetting 2FA on the provider side (website where you set 2FA enabled and which gave you secret and then requests you to input auth code) OR resetting Authenticator application itself?

**Comment by pankiller87** [comment]
I want to reset authenticator password - but i found solution : go in to sd
card an delete data folder of autheticator then i can make new pass and
start from begining :)

W dniu pon., 2.01.2023 o 16:15 Alexander Kopachov ***@***.***>
napisał(a):

> Do you mean resetting 2FA on the provider side (website where you set 2FA
> enabled and which gave you secret and then requests you to input auth code)
> OR resetting Authenticator application?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/akopachov/flipper-zero_authenticator/discussions/43#discussioncomment-4572566>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5BI73UZEZIMLIY6FG2BICDWQLWKHANCNFSM6AAAAAASLGN2CU>
> .
> You are receiving this because you commented.Message ID:
> <akopachov/flipper-zero_authenticator/repo-discussions/43/comments/4572566
> @github.com>
>
-- 
Med venlig hilsen / Best Regards / Pozdrawiam

Marcin Wiszkiel

delta electric
Szczecińska 10, 74-107 Radziszewo
Tel: +48 694 83 1234
Mail: ***@***.***

**Reply by akopachov** [reply]
Yep, this one of the possible options on how to completely reset application with loosing all the changes. Another option is using CLI command `totp reset`.

All of them described in FAQ page ["How to reset application?"](https://github.com/akopachov/flipper-zero_authenticator/wiki/How-to-reset-application%3F)
